Batteries/Data/Vector/Lemmas.lean

```diff
@@ -291,3 +291,15 @@ instance instDecidableExistsVectorZero (P : Vector α 0 → Prop) [Decidable (P
 instance instDecidableExistsVectorSucc (P : Vector α (n+1) → Prop)
     [Decidable (∀ (x : α) (v : Vector α n), ¬ P (v.push x))] : Decidable (∃ v, P v) :=
   decidable_of_iff (¬ ∀ v, ¬ P v) Classical.not_forall_not
+
+instance (α n) [BEq α] [LawfulBEq α] : LawfulBEq (Vector α n) where
+  rfl {a} := by
+    simp only [(· == ·)]
+    rw [mk_isEqv_mk, Array.isEqv_self_beq]
+  eq_of_beq {a b} := by
+    simp only [(· == ·)]
+    rw [mk_isEqv_mk]
+    intro heqv
+    ext
+    have := Array.rel_of_isEqv heqv
```

Comment:
I just noticed that `Array.rel_of_isEqv` is remarkably inconvenient to use. It should be a pair of lemmas:
```
theorem size_eq_size_of_isEqv {v₁ v₂ : Array α} : v₁.isEqv v₂ r → v₁.size = v₂.size := sorry

theorem getElem_rel_getElem_of_isEqv {v₁ v₂ : Array α} {h₁ : i < v₁.size} {h₂ : i < v₂.size} : v₁.isEqv v₂ r → r v₁[i] v₂[i] := sorry
```
What do you think @kim-em ?